### ISSUE TITLE
ValueError: tuple.index(x): x not in tuple in `simulate_petab_amici`

### ISSUE THREAD

**Issue by matthiaskoenig** [AMICI-dev/AMICI#1788]
When running the following petab
[icg_petab_2022-02-05.zip](https://github.com/AMICI-dev/AMICI/files/8603378/icg_petab_2022-02-05.zip)
I get a ValueError. I validated the files with the `petablint` and do not get any errors/warnings.

```python
from pathlib import Path

import petab
from amici.petab_import import import_petab_problem
from amici.petab_objective import simulate_petab


def simulate_petab_amici(yaml_file: Path):
    """Simulate PETab."""

    petab_problem = petab.Problem.from_yaml(yaml_file)
    model = import_petab_problem(petab_problem)
    simulate_petab(petab_problem, model)


if __name__ == "__main__":
    output_dir: Path = Path(__file__).parent / "results" / "icg"
    yaml_file: Path = output_dir / "icg.yaml"
    simulate_petab_amici(yaml_file=yaml_file)
```
Results in 
```bash
Traceback (most recent call last):
  File "/home/mkoenig/git/pkdb_models/pkdb_models/models/icg/fit_petab/icg_amici2.py", line 21, in <module>
    simulate_petab_amici(yaml_file=yaml_file)
  File "/home/mkoenig/git/pkdb_models/pkdb_models/models/icg/fit_petab/icg_amici2.py", line 15, in simulate_petab_amici
    simulate_petab(petab_problem, model)
  File "/home/mkoenig/.virtualenvs/pkdb_models/lib/python3.9/site-packages/amici/logging.py", line 194, in wrapper_timer
    rval = func(*args, **kwargs)
  File "/home/mkoenig/.virtualenvs/pkdb_models/lib/python3.9/site-packages/amici/petab_objective.py", line 133, in simulate_petab
    edatas = create_edatas(
  File "/home/mkoenig/.virtualenvs/pkdb_models/lib/python3.9/site-packages/amici/petab_objective.py", line 521, in create_edatas
    edata = create_edata_for_condition(
  File "/home/mkoenig/.virtualenvs/pkdb_models/lib/python3.9/site-packages/amici/petab_objective.py", line 597, in create_edata_for_condition
    y, sigma_y = _get_measurements_and_sigmas(
  File "/home/mkoenig/.virtualenvs/pkdb_models/lib/python3.9/site-packages/amici/petab_objective.py", line 695, in _get_measurements_and_sigmas
    observable_ix = observable_ids.index(measurement[OBSERVABLE_ID])
ValueError: tuple.index(x): x not in tuple

Process finished with exit code 1
```

**Comment by dweindl** [comment]
In can confirm that this does not work. However, for me it already fails during model import / compilation.

The "problem" is the use of species in `noiseFormula`. This is allowed in PEtab, and `petablint` is right to not complain.
Now the real problems are:
1) that AMICI cannot handle species in the noise model and does not provide a nice error message, but only compiler errors (#919)
2) there is a very unfortunate combination of a) AMICI not supporting states but only observables in noise expressions (#609), and b) PEtab not supporting observables, but only states in the noise expressions (https://github.com/PEtab-dev/PEtab/issues/368). (If you change `Cve_icg` to `model__Cve_icg` in `noiseFormula`, this should be handled correctly by AMICI, but `petablint` will complain.) I noticed that only a few weeks ago, and this is definitely something we need to address.
  This can either be addressed by changing AMICI's PEtab code to transparently add observables for the respective states (or use existing ones, if present). Or propose changing that in PEtab to allow observables instead of states (or in addition), which I think would often be more convenient. (Supporting states there in AMICI would also be an option, but less likely to happen.)

**Comment by dweindl** [comment]
Hi @matthiaskoenig, with AMICI from the `develop` branch (`pip3 install -e git+https://github.com/AMICI-dev/amici.git@develop#egg=amici\&subdirectory=python/sdist`), your noise model should work (#1791).

You will still run into some error, because you have measurements at timepoints where the respective species has concentration of zero, which would lead to division by zero when computing the likelihood. You might want to consider adding an absolute error term.

**Comment by dweindl** [comment]
Should be fixed in v0.11.29

**Comment by matthiaskoenig** [comment]
Thanks so much.
